lib/Sema/TypeCheckStmt.cpp

```diff
@@ -765,10 +765,11 @@ class StmtChecker : public StmtVisitor<StmtChecker, Stmt*> {
 
       // If someone is using an unlabeled break inside of an 'if' or 'do'
       // statement, produce a more specific error.
-      if (S->getTargetName().empty() && !ActiveLabeledStmts.empty() &&
-          (isa<IfStmt>(ActiveLabeledStmts.back()) ||
-           isa<DoStmt>(ActiveLabeledStmts.back())))
+      if (S->getTargetName().empty() &&
+          std::any_of(ActiveLabeledStmts.rbegin(), ActiveLabeledStmts.rend(),
+                      [&](Stmt *S) -> bool { return isa<IfStmt>(S) || isa<DoStmt>(S); })) {
```

Comment:
Nitpick: swiftc style limits lines to 80 columns. Can you reflow this, or use a helper variable?